datalad_next/archive_operations/tarfile.py

```diff
@@ -0,0 +1,93 @@
+"""Archive operation handlers"""
+
+# allow for |-type UnionType declarations
+from __future__ import annotations
+
+import logging
+import tarfile
+from contextlib import contextmanager
+from pathlib import Path
+from typing import (
+    Any,
+    Generator,
+    IO,
+)
+
+# TODO we might just want to do it in reverse:
+# move the code of `iter_tar` in here and have it call
+# `TarArchiveOperations(path).__iter__()` instead.
+# However, the flexibility to have `iter_tar()` behave
+# differently depending on parameters (fp=True/False)
+# is nice, and `__iter__()` only has `self`, such that
+# any customization would need to be infused in the whole
+# class. Potentially cumbersome.
+from datalad_next.iter_collections.tarfile import (
+    TarfileItem,
+    iter_tar,
+)
+
+from . import ArchiveOperations
+from ..config import ConfigManager
```

Comment:
The must not be relative importing across `datalad_next` subpackages. Only within subpackage.

Reply:
Done